zebra-chain/src/sapling/address.rs

```diff
@@ -62,6 +77,22 @@ impl fmt::Display for Address {
     }
 }
 
+impl From<Address> for [u8; 43] {
+    /// Corresponds to the _raw encoding_ of an *Sapling* _shielded payment address_.
+    ///
+    /// <https://zips.z.cash/protocol/protocol.pdf#saplingpaymentaddrencoding>
+    fn from(addr: Address) -> [u8; 43] {
+        use std::convert::TryInto;
+
+        let mut bytes = io::Cursor::new(Vec::new());
+
+        let _ = bytes.write_all(&<[u8; 11]>::from(addr.diversifier));
+        let _ = bytes.write_all(&<[u8; 32]>::from(addr.transmission_key));
+
+        bytes.into_inner().try_into().expect("43 bytes")
```

Comment:
Optional idea: this is similar to the situation above:

```suggestion
        let mut bytes = [0u8; 43];

        let address_bytes: [u8; 11] = addr.diversifier.into();
        let address_end = address_bytes.len();

        bytes[..address_end].copy_from_slice(&address_bytes);
        bytes[address_end..].copy_from_slice(&<[u8; 32]>::from(addr.transmission_key));

        bytes
```